### PR TITLE
[CARBONDATA-1981][Build] Fix compile error in windows env

### PR DESCRIPTION
**Pull request by xuchuanyin**
# Scenario & RootCause

PR #1738 introduced a way to generate version info while building project. It use bash script in maven, which will cause error in Windows environment.

# Modification 

+ Add `windows` profile to support building project in Windows
environment. This profile is disabled by default.

+ If this profile is enabled, a Windows bat will be executed, otherwise a Unix bash script will be executed by default.

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [X] Any interfaces changed?
         `No`
 - [X] Any backward compatibility impacted?
         `No`
 - [X] Document update required?
        `Yes. README in build module is updated`
 - [X] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
         `No, not related`
        - How it is tested? Please attach test report.
         `Not related`
        - Is it a performance related change? Please attach the performance test report.
         `Not related`
        - Any additional information to help reviewers in testing this change.
         `No`
 - [X] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
         `Not related`

### CODE REVIEW

**Comment by xuchuanyin** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2546/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1319/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2713/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2569/

File: core/pom.xml [hunk 1]
```diff
@@ -114,24 +114,22 @@
...
             <configuration>
-              <!-- Execute the shell script to generate the CarbonData build information. -->
-              <target>
-                <exec executable="bash">
-                  <arg value="${project.basedir}/../build/carbondata-build-info"/>
-                  <arg value="${project.build.directory}/extra-resources"/>
-                  <arg value="${project.version}"/>
-                </exec>
-              </target>
+              <executable>${project.basedir}/../build/carbondata-build-info${script.exetension}</executable>
```
Comment (zzcclp) [inline core/pom.xml]: do it need to rename original 'carbondata-build-info' to 'carbondata-build-info.sh'?

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2736/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1336/

File: core/pom.xml (hunk 1 above)
Comment (xuchuanyin) [reply core/pom.xml]: In *nix & Windows environment, maven will execute different scripts, so we need to rename the file.

**Comment by xuchuanyin** [comment]
@zzcclp , do you know how to handle the following `Permission denied` error?

[ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.2.1:exec (default) on project carbondata-core: Command execution failed. Cannot run program "/var/lib/jenkins/workspace/ApacheCarbonPRBuilder/core/../build/carbondata-build-info.sh" (in directory "/var/lib/jenkins/workspace/ApacheCarbonPRBuilder/core"): error=13, Permission denied -> [Help 1]

**Comment by zzcclp** [comment]
@xuchuanyin when rename carbondata-build-info to carbondata-build-info.sh, the permission of it had been changed 755 to 644? so it is permission denied. you can try to chmod arbondata-build-info.sh to 755 and re-commit.

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2742/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1346/

**Comment by zzcclp** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2587/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1352/

**Comment by xuchuanyin** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1370/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2605/

**Comment by xuchuanyin** [comment]
retest this please

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2778/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2617/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1382/

**Comment by zzcclp** [comment]
retest this please.

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2622/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1387/

**Comment by zzcclp** [comment]
retest this please.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1388/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2623/

**Comment by zzcclp** [comment]
retest this please.

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2627/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1392/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1408/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1413/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2644/

**Comment by zzcclp** [comment]
retest this please.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1420/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2652/

**Comment by xuchuanyin** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2671/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1437/

**Comment by chenliang613** [comment]
LGTM, verified in MAC OS and Windows.

**Pull request closed without merging**